src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -22546,6 +22820,54 @@ class Derived : Base, I<(int notA, int notB)>
         }
 
         [Fact]
+        [WorkItem(20528, "https://github.com/dotnet/roslyn/issues/20528")]
+        public void ImplicitBaseImplementationNotConsideredImplementationForInterfaceWithNoNames()
+        {
+            var source = @"
+interface I<T>
+{
+    T M();
+}
+class Base : I<(int a, int b)>
+{
+    public (int a, int b) M() { return (1, 2); }
+}
+class Derived : Base, I<(int, int)>
+{
+    // error
+}
+";
+            var comp = CreateStandardCompilation(source, references: s_valueTupleRefs);
+            comp.VerifyDiagnostics(
+                // (8,27): error CS8141: The tuple element names in the signature of method 'Base.M()' must match the tuple element names of interface method 'I<(int, int)>.M()' (including on the return type).
+                //     public (int a, int b) M() { return (1, 2); }
+                Diagnostic(ErrorCode.ERR_ImplBadTupleNames, "M").WithArguments("Base.M()", "I<(int, int)>.M()").WithLocation(8, 27)
```

Comment:
It looks like this error is reported on the base method which seems surprising, especially since the error does not mention which derived class was involved. I would expect the error to be reported on the type name in `class Derived`.